src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -0,0 +1,168 @@
+package seedu.address.model.session;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Class that handles RecurringSession that extend Session.
+ */
+public class RecurringSession extends Session {
+    public static final String MESSAGE_CONSTRAINTS = "Last date/time does not match starting date and interval.";
+    private Interval interval;
```

Comment:
Hi thanks for the hard work. One of the bugs I found was that if I were to put a value where the end date and start date doesn't match, instead of showing an error message, an exception occurs. I think for this, it would be better to show some form of error message to the user instead.
![image](https://user-images.githubusercontent.com/24240408/112743666-df27e980-8fcb-11eb-9d65-1d072aa0ea22.png)

![image](https://user-images.githubusercontent.com/24240408/112743685-04b4f300-8fcc-11eb-8931-237e9f17ec13.png)

Reply:
Thanks have added this as a new separate issue